File: app/src/androidTest/java/com/github/fo2rist/mclaren/pages/MainPage.java
package com.github.fo2rist.mclaren.pages;


import android.support.test.espresso.ViewInteraction;
import android.support.test.espresso.contrib.DrawerActions;

import com.github.fo2rist.mclaren.R;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.contrib.NavigationViewActions.navigateTo;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

public class MainPage {
    public ViewInteraction onToolbar() {
        return onView(withId(R.id.toolbar));
    }

    public ViewInteraction onContentFrame() {
        return onView(withId(R.id.main_content_frame));
    }

    public ViewInteraction onFloatingButton() {
        return onView(withId(R.id.floatig_button_transmission));
    }

    private ViewInteraction onNavigationView() {
        return onView(withId(R.id.nav_view_main));
    }

    private ViewInteraction onFooterNavigationView() {
        return onView(withId(R.id.nav_view_footer));
    }

    public ViewInteraction onMenuNewsFeed() {
        return onView(withText("Feed"));
    }

    public ViewInteraction onMenuStories() {
        return onView(withText("Stories"));
    }

    public ViewInteraction onMenuCircuits() {
        return onView(withText("Circuits"));
    }

    public ViewInteraction onMenuDrivers() {
        return onView(withText("Drivers"));
    }

    public ViewInteraction onMenuCar() {
        return onView(withText("Car"));
    }

    public ViewInteraction onOptionMenuAbout() {
        return onView(withText("About"));
    }

    public void openNavigationDrawer() {
        onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
    }

    public void navigateToMenuItem(int menuItemId) throws InterruptedException {
        openNavigationDrawer();
        onNavigationView()
                .perform(navigateTo(menuItemId));
    }

    public void navigateToFooterMenuItem(int menuItemId) throws InterruptedException {
        openNavigationDrawer();
        onFooterNavigationView()
                .perform(navigateTo(menuItemId));
    }

    public ViewInteraction onMenuOfficialSite() {
        return onView(withText("Official site"));
    }
}

File: app/src/androidTest/java/com/github/fo2rist/mclaren/pages/MainPage.kt
package com.github.fo2rist.mclaren.pages


import com.agoda.kakao.common.views.KView
import com.agoda.kakao.drawer.KDrawerView
import com.agoda.kakao.navigation.KNavigationView
import com.agoda.kakao.screen.Screen
import com.github.fo2rist.mclaren.R

class MainPage : Screen<MainPage>() {

    val menuStories: KView = KView { withText(R.string.navigation_item_stories) }

    val menuCircuits: KView = KView { withText(R.string.navigation_item_season_calendar) }

    val menuLandoTwitter: KView = KView { withText(R.string.navigation_item_lando_twitter) }

    val menuDanielTwitter: KView = KView { withText(R.string.navigation_item_daniel_twitter) }

    val menuTeamTwitter: KView = KView { withText(R.string.navigation_item_team_twitter) }

    val menuDriver1Twitter: KView = KView { withText(R.string.navigation_item_lando_twitter) }

    val menuDriver2Twitter: KView = KView { withText(R.string.navigation_item_daniel_twitter) }

    val menuDrivers: KView = KView { withText(R.string.navigation_item_drivers) }

    val menuCar: KView = KView { withText(R.string.navigation_item_car) }

    val menuOfficialSite: KView = KView { withText(R.string.navigation_item_official_site) }

    val optionMenuAbout: KView = KView { withText(R.string.action_about) }

    val transmissionButton: KView = KView { withId(R.id.floating_button_transmission) }

    val upcomingEventButton: KView = KView { withId(R.id.floating_button_upcoming_event) }

    fun openNavigationDrawer() = KDrawerView { withId(R.id.drawer_layout) }.open()

    fun navigateToMenuItem(menuItemId: Int) {
        openNavigationDrawer()
        navigationView.navigateTo(menuItemId)
    }

    private val navigationView = KNavigationView { withId(R.id.nav_view_main) }

    fun navigateToFooterMenuItem(menuItemId: Int) {
        openNavigationDrawer()
        footerNavigationView.navigateTo(menuItemId)
    }

    private val footerNavigationView = KNavigationView { withId(R.id.nav_view_footer) }
}
